README.md

```diff
@@ -228,3 +228,125 @@ This endpoint returns JSON. An example follows below.
 
 Note that some of the keys may be missing if access to them was not requested
 and granted.
+
+## Webhooks
+
+### Description
+
+Webhooks allow you to build or set up Applications which subscribe to certain events on Fractal ID. When one of those events is triggered, we'll send a HTTP `POST` payload to the webhook's configured URL.
+
+### Terminology
+
+- Callback URL - the URL of the server that will receive the webhook `POST` requests.
+- Secret token - token that allows you to ensure that `POST` requests sent to the callback URL are from Fractal.
+- Notification - webhooks `POST` request which is triggered by certain events on Fractal ID.
+
+### Setup
+
+The integration of Webhooks is currently available for select partners.
+
+Setup involves:
+1. Partner providing Fractal with callback URL and available notification types it wants to subscribe.
+1. Fractal providing the partner with a secret token.
+
+### Available notifications types
+
+#### Verification approved
+
+Once user is approved, the partner can get notified about user verification process being successfully completed. Upon getting `verification_approved` notification, partner can use specified user access token and retrieve latest information about the user and/or perform its business logic.
+
+Example payload:
+
+```
+{
+  type: "verification_approved",
+  data: {
+    user_id: "14ec6af0-12f8-4bce-a6ab-01ce87fa1812",
+  }
+}
+```
+
+### Securing webhooks
+
+#### Callback URL
+
+Fractal ID accepts the following callback URLs:
+* Secure sites (HTTPS) - for production usage.
+* Localhost URLs - for development / testing. For example: `http://localhost:3000/callback`
+
+#### Setting your secret token
+
+After Fractal provides secret token to the partner, it should set up an environment variable on its server that stores this token. Typically, this is as simple as running:
+
+```
+export SECRET_TOKEN=your_token
+```
+
+**Never** hardcode the token into your app!
+
+#### Validating payloads from Fractal ID
+
+When your secret token is set, Fractal ID uses it to create a hash signature with each payload. The hash signature is passed along with each request in the headers as `X-Fractal-Signature`. Fractal ID uses HMAC hexdigest to compute the hash.
+
+Partner should verify the signature on their server. Example implementation in Ruby:
+
+```
+
+def verify_signature
+  payload_body = request.body.read
+  signature = "sha1=" + OpenSSL::HMAC.hexdigest(OpenSSL::Digest.new("sha1"), ENV["SECRET_TOKEN"], payload_body)
+
+  return if Rack::Utils.secure_compare(signature, request.headers["X-Fractal-Signature"])
+
+  render json: { error: "signature_mismatch" }, status: 500
```

Comment:
I would `status: 400` instead, since it's a client error, not a server error.

Reply:
Hmm I was also considering other response code. Github example suggests returning 500. 🤔 

If the endpoint does not exist, servers usually return 500. Returning 400 would expose endpoint being present. And only Fractal ID should call this endpoint.

I am on the fence with this one. Changing for now.